source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -789,7 +789,7 @@ void ShaderGraph::optimize(GenContext& context)
     size_t numEdits = 0;
     for (ShaderNode* node : getNodes())
     {
-        if (!node->hasClassification(ShaderNode::Classification::DO_NOT_OPTIMIZE) && node->hasClassification(ShaderNode::Classification::CONSTANT))
+        if (node->hasClassification(ShaderNode::Classification::CONSTANT))
```

Comment:
This is much clearer to read. I assume we could add an option in GenOptions to turn off constant optimizations at some point so we can edit shader inputs without recompiles.